### DISCUSSION TITLE
install devDependencies only for current project

### DISCUSSION THREAD

**Discussion by imshenshen** [pnpm/pnpm#16035] [Ideas]
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

**Describe the user story**

when run `pnpm install`, it install `devDependencies of devDependencies` of my project, the `devDependencies of devDependencies` usually useless for development. 

**Describe the solution you'd like**

give a option to only install `devDependencies of 'my project'`

**Comment by frank-dspeed** [comment]
at present i see this already do able via some folders where you configure diffrent workspaces with diffrent settings

you can simply create folders and link your main folders as your system already supports filesystem links maybe that helps you else post a link to your project and i do a PR

**Comment by zkochan** [comment]
I am not sure what you mean as dev deps of dev deps are not installed. Try

```
pnpm add is-odd -D
```

See that mocha is not getting installed.

**Comment by frank-dspeed** [comment]
@zkochan his goal is to have something like Install dev only without dependencies. 

package.json
```
devDeps
deps
```

install only the once under devDeps without the dev deps of them of course

**Comment by frank-dspeed** [comment]
i did come up with new patterns for 2026 devDependencys should not be used or exist you simple create a extra dev-package under packages/ in your workspaces this dev package gets shared for all your packages and replaces flexible dev dependencys share maintain and install able when needed cheers hope it makes sense
